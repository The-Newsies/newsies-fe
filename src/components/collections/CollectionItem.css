$bg-color: #E6E5E1;
$shdow-gray: #d3d3d5;
$orange: #DA4D1D;
$blue: #2b26c3;

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

h2{
  pointer-events:none;
}

.container {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  .card {
    position: relative;
    width: 400px;
    height: 200px;
    background-color: #fff;
    overflow: hidden;
    margin: 15px 8px;
    box-shadow: 0 1px 3px rgba(60, 60, 60, 0.12), 0 1px 2px rgba(60, 60, 60,0.24);
    transition: all 0.3s cubic-bezier(.25,.8,.25,1);

    .text {
      display: flex;
      justify-content: center;
      flex-direction: row;
      align-items: center;
      flex-direction: column;
      height: 100%;
    }

    &:before {
      content: "";
      z-index: 1;
      position: absolute;
      top: -10px;
      left: 32px;
      display: block;
      width: 18px;
      height: 18px;
      border-radius: 20px;
      background-color: $bg-color;
    }
    &:after {
      content: "";
      z-index: 0;
      position: absolute;
      bottom: -10px;
      left: 32px;
      display: block;
      width: 18px;
      height: 18px;
      border-radius: 20px;
      background-color: $bg-color;
    }
    ul {
      z-index: 99;
      position: absolute;
      left: 39px;
      top: 5px;
      list-style-type: none;
      li {
        width: 2px;
        height: 2px;
        border-radius: 2px;
        margin: 6px 0;
        background-color: $bg-color;
      }
    }
    h3 {
      z-index: 99;
      font-family: "Poppins", sans-serif;
      bottom: 0;
      font-size: 20px;
      font-weight: 700;
      text-align: center;
      color: #000;
    }
    p {
      z-index: 100;
      top: 40px;
      padding-left: 10px;
      color: #333;
      opacity: 0.8;
      font-size: 14px;
      letter-spacing: 1px;
      /*writing-mode: vertical-lr;*/
      -webkit-transition: all 0.2s ease;
      transition: all 0.2s ease;
    }

    a {
      text-decoration: none;
    }

    .deleteButton {
      filter: grayscale(0);
      display: flex;
      align-items: center;
      
      
    }

    .deleteButton button {
      font-size: 1rem;
      font-weight: bolder;
      position: absolute;
      right: 10px;
      bottom: 15px;
      width: 30px;
      height: 30px;
      background-color: red;
      border: none;
      border-radius: 30px;
      cursor: pointer;
      outline:none;
      transition: all 0.3s ease;


      i {
        font-size: 3rem;
      }
    }

    &:hover button {
      transform: scale(5.5);

    }

    &:hover p {
      color: #2b26c3;
    }

    &:hover h3 {
      color: #2b26c3;
    }
  }
}

.title {
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: center;
  padding: .5rem;
  height: 6rem;
  overflow: auto;
}

.card:hover {
  box-shadow: 0 14px 28px rgba(70, 70, 70,0.25), 0 10px 10px rgba(70,70,70,0.22);
}

