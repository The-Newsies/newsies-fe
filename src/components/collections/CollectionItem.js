import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './CollectionItem.css';


export default function CollectionItem({ name, description, _id, handleDelete }) {
  return (
    <div className={styles.container}>
      <div className={styles.card}>
        <div className={styles.text}>
          <Link to={`/collection/${_id}`}>
            <h3>{name}</h3>
            <p>{description}</p>
          </Link>
        </div>
        <div className={styles.deleteButton}>
          <button onClick={() => handleDelete(_id)}>âœ—</button>
        </div>
      </div>
    </div>
  );
}

CollectionItem.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  _id: PropTypes.string,
  handleDelete: PropTypes.func.isRequired
};
