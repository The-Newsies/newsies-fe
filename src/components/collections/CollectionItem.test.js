import React from 'react';
import { shallow } from 'enzyme';
import CollectionItem from './CollectionItem';

describe('CollectionItem component', () => {
  it('renders CollectionItem', () => {
    const wrapper = shallow(<CollectionItem 
      name='Title'
      description='description'
      _id='_id'
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
