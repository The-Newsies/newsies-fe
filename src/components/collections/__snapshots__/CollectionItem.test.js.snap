// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component renders CollectionItem 1`] = `
<Fragment>
  <Link
    to="/collection/delete"
  >
    <h3>
      Title
    </h3>
    <p>
      description
    </p>
  </Link>
  <button
    onClick={[Function]}
  >
    âœ—
  </button>
</Fragment>
`;
